internal/common/states.go

```diff
@@ -48,7 +48,7 @@ func unmarshalStateHelper(data []byte, mapping []string) (int, error) {
 	if err != nil {
 		return 0, err
 	}
-	for n, str := range getStateMapping() {
+	for n, str := range mapping {
```

Comment:
Hehe, good one! :-D given that line 61 seems to be the only caller, this should be fine. But I wonder why wasn't this caught by the compiler or the linter...

Note about the commit message: We usually start with the name of module/package/file we modify. So instead of a "fix" you can use something like "common" or "internal/common". And don't hesitate to write long commit messages explaining even the smallest change. The best commits are those with longer commit message than the patch itself :-D

Reply:
Agreed that it's surprising that the linter didn't pick it up.

Thanks for the info on the commit messages. I'll keep that in mind for future commits and PRs